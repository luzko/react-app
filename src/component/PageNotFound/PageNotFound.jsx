import React from 'react';
import style from './PageNotFound.module.css';
import notFound from '../../../public/404.png';
import {Link} from 'react-router-dom';

const PageNotFound = () => {
  return (
      <div className={style.pageNotFound}>
        <span className={style.text}>Page Not Found</span>
        <img
            src={notFound}
            alt='not found'
            className={style.notFoundImage}
        />
        <Link to='/'>
          <button className={style.goBack}>GO BACK</button>
        </Link>
      </div>
  );
};

export default PageNotFound;
