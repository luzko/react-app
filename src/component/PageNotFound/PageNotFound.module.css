.pageNotFound {
  min-height: 800px;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: #232323;
}

.text {
  font-size: 32px;
  color: #ffffff;
  margin-top: 5vh;
  margin-bottom: 5vh;
}

.notFoundImage {
  height: 300px;
}

.goBack {
  width: 200px;
  height: 50px;
  border-radius: 5px;
  border: solid 1px #f65261;
  font-size: 18px;
  color: #f65261;
  background-color: #232323;
  margin: 10px;
}

.logo {
  width: 110px;
}
