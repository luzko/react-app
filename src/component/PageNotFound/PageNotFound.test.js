import React from 'react';
import renderer from 'react-test-renderer';
import PageNotFound from './PageNotFound';
import { MemoryRouter } from 'react-router-dom';

test('snapshot test for page not found', () => {
  const component = renderer.create(
      <MemoryRouter>
        <PageNotFound />
      </MemoryRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
