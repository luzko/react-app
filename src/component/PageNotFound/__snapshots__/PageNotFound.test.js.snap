// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for page not found 1`] = `
<div
  className="pageNotFound"
>
  <span
    className="text"
  >
    Page Not Found
  </span>
  <img
    alt="not found"
    className="notFoundImage"
    src="404.png"
  />
  <a
    href="/"
    onClick={[Function]}
  >
    <button
      className="goBack"
    >
      GO BACK
    </button>
  </a>
</div>
`;
